crates/environ/src/stack_map.rs

```diff
@@ -57,4 +128,26 @@ impl StackMap {
             ptr_to_gc_ref.cast::<u32>()
         })
     }
+
+    /// Returns the offsets that this stack map registers GC references at.
+    pub fn offsets(&self) -> impl Iterator<Item = u32> + '_ {
+        // Here `self.data` is a bit set of offsets divided by 4, so iterate
+        // over all the bits in `self.data` and multiply their position by 4.
+        let bit_positions = self.data.iter().enumerate().flat_map(|(i, word)| {
+            let i = i as u32;
+            let mut remaining = word.get(LittleEndian);
+            core::iter::from_fn(move || {
+                if remaining == 0 {
+                    None
+                } else {
+                    let pos = remaining.trailing_zeros();
+                    let bit = 1 << pos;
+                    remaining ^= bit;
+                    Some(i * 32 + pos)
+                }
+            })
+        });
+
+        bit_positions.map(|pos| pos * 4)
```

Comment:
The `CompoundBitSet::ref_from_raw` thing would help tidy up some of this stuff too, so that it doesn't have to have a copy of the bitset logic inline here.